src/main/java/seedu/address/ui/RestaurantCard.java

```diff
@@ -60,21 +62,18 @@ public RestaurantCard(Restaurant restaurant, int displayedIndex) {
         cuisine.setText(restaurant.getCuisine().cuisine);
         String visited;
         if (restaurant.getVisit().visit.equals("Yes")) {
-            visited = "Visited";
+            visit.setText("Visited");
+            visit.setBackground(new Background(new BackgroundFill(Color.rgb(30, 170, 50),
+                    CornerRadii.EMPTY, Insets.EMPTY)));
+
         } else {
-            visited = "Not visited";
-        }
-        visit.getChildren().add(new Label(visited));
-        String remarkValue = "";
-        int i = 0;
-        while (i < restaurant.getRemark().size()) {
-            remarkValue += restaurant.getRemark().get(i).value;
-            if (i != restaurant.getRemark().size() - 1) {
-                remarkValue += ", ";
-            }
-            i++;
+            visit.setText("Not Visited");
+            visit.setBackground(new Background(new BackgroundFill(Color.rgb(114, 13, 40),
+                    CornerRadii.EMPTY, Insets.EMPTY)));
+
```

Comment:
Why you don't want to setText first then decide on colour can have one less visited.setText() cause if you do it this way not sure whether they will say got unnecessary repeated code O: HAHA

Reply:
No I cannot because I am setting visit to different values here